src/Symfony/Component/HttpKernel/DependencyInjection/RegisterLocaleAwareServicesPass.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\DependencyInjection;
+
+use LogicException;
+use Symfony\Component\DependencyInjection\Argument\IteratorArgument;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Contracts\Service\LocaleAwareInterface;
+
+/**
+ * Register all services that implements LocaleAwareInterface into the listener.
+ *
+ * @author Pierre Bobiet <pierrebobiet@gmail.com>
+ */
+class RegisterLocaleAwareServicesPass implements CompilerPassInterface
+{
+    private $listenerServiceId;
+    private $localeAwareTag;
+
+    public function __construct(
+        string $listenerServiceId = 'locale_aware_listener',
+        string $localeAwareTag = 'locale_aware'
+    ) {
+        $this->listenerServiceId = $listenerServiceId;
+        $this->localeAwareTag = $localeAwareTag;
+    }
+
+    public function process(ContainerBuilder $container)
+    {
+        if (false === $container->hasDefinition($this->listenerServiceId)) {
+            return;
+        }
+
+        $services = array();
+
+        foreach ($container->findTaggedServiceIds($this->localeAwareTag) as $id => $tags) {
+            $definition = $container->getDefinition($id);
+            $serviceClass = $container->getParameterBag()->resolveValue($definition->getClass());
+
+            if (!is_a($serviceClass, LocaleAwareInterface::class, true)) {
```

Comment:
this check is not needed - it should be removed to prevent forcing unneeded classes at compile time